ddtrace/tracer/writer.go

```diff
@@ -89,22 +89,38 @@ func (h *agentTraceWriter) flush() {
 	h.payload = newPayload()
 	go func(p *payload) {
 		defer func(start time.Time) {
+			// Once the payload has been read, clear the buffer for garbage collection to avoid
+			// a memory leak when references to this object may still be kept by faulty transport
+			// implementations or the standard library. See dd-trace-go#976
+			p.buf = bytes.Buffer{}
+			p.reader = nil
```

Comment:
Let's make this a `payload` method. We don't want to be messing with the payload internals here.